.github/workflows/chef-test.yml

```diff
@@ -0,0 +1,86 @@
+name: chef-test
+
+# Only run tests for main branch or if the PR has relevant changes
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      - '.github/workflows/chef-test.yml'
+      - 'deployments/chef/**'
+      - 'internal/buildscripts/packaging/tests/**'
+
+concurrency:
+  group: chef-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHON_VERSION: 3.8.5
+  PIP_VERSION: 20.2.4
+  REQUIREMENTS_PATH: "internal/buildscripts/packaging/tests/requirements.txt"
+  RESULT_PATH: "~/testresults"
+
+jobs:
+  chef-lint-spec-test:
+    name: chef-lint-spec-test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+      
+      - name: lint
+        run: make -C deployments/chef lint
+
+      - name: Rake spec
+        run: |
+          make -C deployments/chef rake-spec
+
+  chef-test:
```

Comment:
Hi @jcheng-splunk
I have added windows `pytest` scripts for the windows CI test. 
Note: 
- Here I am spinning up Windows docker image to run the chef test because by doing that we don’t have to take care of removing files and packages installed by the previous test (On every pytest we have new windows image on which we are deploying collector using chef)
- By considering your latest commit I have also added Dockerfile to run pytest on the Windows server 2022 version
- Docker image of Windows server 2012 is not available so I am not considering it in the windows test